packages/frontapp/src/app/components/SearchByUserName/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react';
+import { useAppSelector } from '../../hooks/useTypedSelector';
+import { useDebounce } from '../../hooks/useDebounce';
+import { useAppDispatch } from '../../hooks/useTypedDispatch';
+import { useNavigate } from 'react-router-dom';
+import { useGetAllPersonsQuery } from '@mimir/apollo-client';
+import { RoutesTypes } from '../../../utils/routes';
+import { setActiveTab } from '../../store/slices/tabsSlice';
+import { t } from 'i18next';
+import { InputSearch, StyledSearchIcon, WrapperInput } from '../Search';
+import { setSearchReaders } from '../../store/slices/readersSlice';
+
+const SearchByUserName = () => {
```

Comment:
This looks like a pretty generic search components. It makes sense to refactor it and split into `SearchInput` and `SearchByWhatever`